l5kit/l5kit/evaluation/metrics.py

```diff
@@ -0,0 +1,46 @@
+import numpy as np
+
+
+def neg_multi_log_likelihood(
+    gt: np.ndarray, pred: np.ndarray, confidences: np.ndarray, avails: np.ndarray
+) -> np.ndarray:
+    """
+    Compute a negative log-likelihood for the multi-modal scenario.
+    log-sum-exp trick is used here to avoid underflow and overflow, For more information about it see:
+    https://en.wikipedia.org/wiki/LogSumExp#log-sum-exp_trick_for_log-domain_calculations
+    https://timvieira.github.io/blog/post/2014/02/11/exp-normalize-trick/
+    https://leimao.github.io/blog/LogSumExp/
+
+    Args:
+        gt (np.ndarray): array of shape (time)x(2D coords)
+        pred (np.ndarray): array of shape (modes)x(time)x(2D coords)
+        confidences (np.ndarray): array of shape (modes) with a confidence for each mode in each sample
+        avails (np.ndarray): array of shape (time) with the availability for each gt timestep
+    Returns:
+        np.ndarray: negative log-likelihood for this example, a single float number
+    """
+    assert len(pred.shape) == 3, f"expected 3D (MxTxC) array for pred, got {pred.shape}"
+    num_modes, future_len, num_coords = pred.shape
+
+    assert gt.shape == (future_len, num_coords), f"expected 2D (TxC) array for gt, got {gt.shape}"
+    assert confidences.shape == (num_modes,), f"expected 1D (M) array for gt, got {confidences.shape}"
```

Comment:
Maybe it would help here to say what M T and C are in the error messages. Will help  contestants iterate quicker maybe. Like: "expected 1D (num_modes) array for gt, got {confidences.shape}"

Reply:
Let's see if we can manage to fit them in without breaking all lines